src/atlas/mesh/MeshBuilder.cc

```diff
@@ -43,16 +142,41 @@ Mesh MeshBuilder::operator()(const std::vector<double>& lons, const std::vector<
     return operator()(nb_nodes, lons.data(), lats.data(), ghosts.data(), global_indices.data(), remote_indices.data(),
                       remote_index_base, partitions.data(), nb_tris,
                       reinterpret_cast<const gidx_t*>(tri_boundary_nodes.data()), tri_global_indices.data(), nb_quads,
-                      reinterpret_cast<const gidx_t*>(quad_boundary_nodes.data()), quad_global_indices.data());
+                      reinterpret_cast<const gidx_t*>(quad_boundary_nodes.data()), quad_global_indices.data(),
+                      config);
 }
 
 Mesh MeshBuilder::operator()(size_t nb_nodes, const double lons[], const double lats[], const int ghosts[],
                              const gidx_t global_indices[], const idx_t remote_indices[], const idx_t remote_index_base,
                              const int partitions[], size_t nb_tris, const gidx_t tri_boundary_nodes[],
                              const gidx_t tri_global_indices[], size_t nb_quads, const gidx_t quad_boundary_nodes[],
-                             const gidx_t quad_global_indices[]) const {
+                             const gidx_t quad_global_indices[],
+                             const eckit::Configuration& config) const {
+    // Get MPI comm from config name or fall back to atlas default comm
+    const eckit::mpi::Comm& comm = (config.has("mpi comm name") ?
+                                    eckit::mpi::comm(config.getString("mpi comm name").c_str())
+                                    : atlas::mpi::comm());
```

Comment:
We could clean this a bit more:
```suggestion
    auto mpi_comm_name = [](const auto& config) {
        return  config.getString("mpi_comm", atlas::mpi::comm().name()).c_str();
    };
    const eckit::mpi::Comm& comm = eckit::mpi::comm(mpi_comm_name(config));
```